Doberman/AlarmNode.py

```diff
@@ -126,3 +131,30 @@ def process(self, packages):
                 msg += f' {low:.3g} to {high:.3g}.'
             self.log_alarm(msg, packages[-1]['time'])
 
+
+class IntegerAlarmNode(AlarmNode):
+    """
+    Integer status quantities are a fundamentally different thing from physical values.
+    It makes sense to process them differently. The thresholds should be stored as [value, message] pairs.
+    """
+    def process(self, package):
+        value = int(package[self.input_var])
+        for v, msg in self.config['alarm_thresholds']:
+            if value == int(v):
+                self.log_alarm(f'Alarm for {self.description}: {msg}')
+                break
+
+
+class BitmaskIntegerAlarmNode(AlarmNode):
+    """
+    Sometimes the integer represents a bitmask.
+    """
+    def process(self, package):
+        value = int(package[self.input_var])
+        alarm_msg = []
+        for i, msg in self.config['alarm_thresholds']:
+            if value & (1 << int(i)):
```

Comment:
Assumes that the bit being 1 is bad

Reply:
I hadn't considered this, but it's absolutely something to add.

Reply:
New method here: the config holds a list of (bitmask, value, msg) entries. This allows multiple bits to be checked for being either 1 or 0 simultaneously.

Reply:
I like it